others/sparse_matrix_with_vectors.cpp

```diff
@@ -0,0 +1,128 @@
+ * @param vect Matrix to be tested
+ * @returns zero if matrix is not sparse
+***/
+bool is_sparse_matrix(std::vector<std::vector<int>> vect)
+{
+    int zero_counter = 0;
+    for(int i =0; i<vect.size(); i++)
+    {
+        for(int j = 0; j<vect[i].size(); j++)
+        {
+            if (vect[i][j]==0)
+            {
+                zero_counter+=1;
+            }
+        }
+    }
+
+    return (zero_counter>(vect.size() * vect[0].size())* 0.5); //return 1 if zero values are more than non zero values
+
+}
+
+/*******************************************************************************
+ * @brief 
+ * function to display the Matrix elements
+ * @param vect Matrix to be displayed
+ * @returns void
+*******************************************************************************/
+
+void display(std::vector<std::vector<int>> vect)
+{
+    for(int i =0; i<vect.size(); i++)
+    {
+        for(int j = 0; j<vect[i].size(); j++)
+        {
+            std::cout<<vect[i][j]<<" ";
+        }
+        std::cout<<"\n";
+    }
+}
+
+/*******************************************************************************
+ * @brief Self-test implementation #1
+ * @returns void
+ *******************************************************************************/
+
+void test1()
+{
+    std::vector<std::vector<int>> vect1 = {{1, 0, 0, 1}, {0, 1, 0, 2}, {1, 0, 1, 4}};
+    int expected_ans = 0;
+    int derived_ans = is_sparse_matrix(vect1);
+    std::cout<<"Test #1: ";
+    assert(derived_ans == expected_ans);
+    std::cout<<"Passed! \n"; 
+
+}
+
+/*******************************************************************************
+ * @brief Self-test implementation #2
+ * @returns void
+ *******************************************************************************/
+
+void test2()
+{
+    std::vector<std::vector<int>> vect1 = {{1, 1, 1, 0}, {0, 1, 0, 2}, {1, 1, 1, 1}};
+    int expected_ans = 0;
+    int derived_ans = is_sparse_matrix(vect1);
+    std::cout<<"Test #2: ";
+    assert(derived_ans == expected_ans);
+    std::cout<<"Passed! \n"; 
+
+}
+/*******************************************************************************
+ * @brief Self-test implementation #3
+ * @returns void
+ *******************************************************************************/
+
+void test3()
+{
+    std::vector<std::vector<int>> vect1 = {{0, 0, 0, 1}, {0, 0, 0, 2}, {0, 1, 0, 4}};
+    int expected_ans = 1;
+    int derived_ans = is_sparse_matrix(vect1);
+    std::cout<<"Test #3: ";
+    assert(derived_ans == expected_ans);
+    std::cout<<"Passed! \n"; 
+
+}
+
+ /*******************************************************************************
+     * @brief Main function
+     * @returns 0 on exit
+     *******************************************************************************/
+
+int main()
+{
+    test1();  // run self-test implementation #1
+    test2();  // run self-test implementation #2
+    test3();  // run self-test implementation #3
```

Comment:
should consolidate these tests into a single test
i.e
```c++
tests() {
   some_specific_case();
   some_other_case();
}
```
and in main
```cpp
tests() // run self-test implementations
```
also adding a different function for each case which are not that unique seems a bit redundant dont you think?